backend/src/packages/chaiNNer_pytorch/pytorch/processing/image_align_rife.py

```diff
@@ -0,0 +1,211 @@
+
+# Original Rife Frame Interpolation by hzwer
+# https://github.com/megvii-research/ECCV2022-RIFE
+# https://github.com/hzwer/Practical-RIFE
+
+# Modifications to use Rife for Image Alignment by tepete/pifroggi ('Enhance Everything!' Discord Server)
+
+# Additional helpful github issues
+# https://github.com/megvii-research/ECCV2022-RIFE/issues/278
+# https://github.com/megvii-research/ECCV2022-RIFE/issues/344
+
+import sys
+import os
+
+current_dir = os.path.dirname(os.path.abspath(__file__))
+project_root = os.path.dirname(current_dir)
+sys.path.append(project_root)
+
+import torch
+import torch.nn.functional as F
+import numpy as np
+import cv2
+import requests
+import zipfile
+from pathlib import Path
+from api import NodeContext
+from nodes.properties.inputs import ImageInput, EnumInput, NumberInput, BoolInput, SliderInput
+from nodes.properties.outputs import ImageOutput
+from ...settings import PyTorchSettings, get_settings
+from .. import processing_group
+from nodes.impl.pytorch.utils import np2tensor, tensor2np
+from nodes.impl.resize import resize, ResizeFilter
+from nodes.utils.utils import get_h_w_c
+from packages.chaiNNer_pytorch.pytorch.processing.rife.IFNet_HDv3_v4_14_align import IFNet
+from enum import Enum
+
+class PrecisionMode(Enum):
+    FIFTY_PERCENT         = 2000
+    ONE_HUNDRED_PERCENT   = 1000
+    TWO_HUNDRED_PERCENT   = 500
+    FOUR_HUNDRED_PERCENT  = 250
+    EIGHT_HUNDRED_PERCENT = 125
+
+def calculate_padding(height, width, precision_mode):
+    if precision_mode == PrecisionMode.EIGHT_HUNDRED_PERCENT:
+        pad_value = 4
+    elif precision_mode == PrecisionMode.FOUR_HUNDRED_PERCENT:
+        pad_value = 8
+    elif precision_mode == PrecisionMode.TWO_HUNDRED_PERCENT:
+        pad_value = 16
+    elif precision_mode == PrecisionMode.ONE_HUNDRED_PERCENT:
+        pad_value = 32
+    else:
+        pad_value = 64
+    
+    pad_height = (pad_value - height % pad_value) % pad_value
+    pad_width = (pad_value - width % pad_value) % pad_value
+    return pad_height, pad_width
+
+def download_model(download_url, model_path, model_file, zip_inner_path):
+    model_dir = Path(model_path)
+    model_dir.mkdir(parents=True, exist_ok=True)
+    zip_path = model_dir / "model.zip"

```

Comment:
Instead of this whole sequence you do here, I would prefer if you did something like this: https://github.com/chaiNNer-org/chaiNNer/blob/bfaac0a516057d0898154618b3b28b6109a4eafa/backend/src/packages/chaiNNer_pytorch/pytorch/restoration/upscale_face.py#L169

This gets the appdata/roaming path where we store everything else. The code currently just puts it near wherever chaiNNer is running.